pyfar/classes/filter.py

```diff
@@ -292,6 +292,14 @@ def impulse_response(self, n_samples):
         impulse_response : Signal
             The impulse response of the filter.
         """
+        # set or check the impulse response length
+        impulse_response_length = int(np.max(self.impulse_response_length()))
+        if n_samples is None:
+            n_samples = impulse_response_length
```

Comment:
seems like n_samples has a defualt, but its not mentioned in the docs

Reply:
True, I added it to the docs. Note that this method is overwritten in each derived class and that this docstring is not shown in the docs for this reason. But still better to have a clean internal documentation :)